addTimeShift_function.R

```diff
@@ -0,0 +1,69 @@
+require(lubridate)
+require(dplyr)
+
+#' Add column of time-shifted shifted metrics
+#' 
+#' This function takes a dataframe with a date column and a metric column,
+#' along with any other dimensions in the dataset, and creates a new
+#' metric column time-shifted by the parameters specified. E.g. clicks for
+#' this row one year ago or one week ago.
+#' 
+#' @param data.in The dataframe containing a date and metric
+#' @param date The date column in the dataframe
+#' @param n.per The period of time to shift by (day, week, month, year)
+#' @param n.shift The number of period shifts to apply
+#' @export
+
+addTimeShift <- function(data.in,
+                         metric,
+                         date.col = "date",
+                         time.shift = 1,
+                         shift.per = 364) {
```

Comment:
Baring confusion from my unhelpful comments, it's actually pretty straightforward.

364 is the default b/c what you actually want is 52 weeks * 7 days ago for a Y/Y comparison -- otherwise your Y/Y growth curves will have some strange DOW effects.

time.shift and n.shift are the same -- I just didn't update the documentation. 

shift.per (aka n.per) is the period to compare while time.shift is the number of periods to compare. E.g. using words, a shift.per of 1 week and a time.shift of 3 would show you today's values overlaid with the values of 1 week ago, 2 weeks ago, and 3 weeks ago.